core/src/main/scala/weaponregex/parser/ParserJVM.scala

```diff
@@ -44,6 +48,39 @@ class ParserJVM private[parser] (pattern: String) extends Parser(pattern) {
     Indexed("""\0""" ~ (CharIn("0-3") ~ CharIn("0-7").rep(exactly = 2) | CharIn("0-7").rep(min = 1, max = 2)).!)
       .map { case (loc, octDigits) => MetaChar("0" + octDigits, loc) }
 
+  /** Parse a single literal character that is allowed to be in a character class
+    * @return [[weaponregex.model.regextree.Character]] tree node
+    * @example `"{"`
+    * @see [[weaponregex.parser.Parser.charClassSpecialChars]]
+    */
+  override def charClassCharLiteral[_: P]: P[Character] =
+    Indexed(CharPred(!charClassSpecialChars.contains(_)).! | "&".! ~ !"&")
+      .map { case (loc, c) => Character(c.head, loc) }
+
+  /** Parse a sequence of character class items as a 'naked character class'. It is used only inside the character class intersection.
+    *
+    * @return [[weaponregex.model.regextree.CharacterClassNaked]] tree node
+    */
+  def charClassNaked[_: P]: P[CharacterClassNaked] = Indexed(classItem.rep(minCharClassItem))
+    .map { case (loc, nodes) => CharacterClassNaked(nodes, loc) }
+
+  /** Parse a character class intersection used inside a character class.
+    *
+    * @return [[weaponregex.model.regextree.CharClassIntersection]] tree node
+    * @example `"abc&&[^bc]&&a-z"`
+    */
+  def charClassIntersection[_: P]: P[Seq[CharClassIntersection]] =
+    Indexed(charClassNaked.rep(2, sep = "&&"))
+      .map { case (loc, nodes) => Seq(CharClassIntersection(nodes, loc)) }
```

Comment:
Shouldn't `charClassIntersection` return just a `CharClassIntersection` instead of a `Seq[CharClassIntersection]`?

Reply:
this way it can be easily used by charClass, because it needs a list of children. You could also make it return not a list, but then you would need some pattern matching in charClass.

Reply:
Sure, can you give an example of how you would implement it? I think I would create a monstrosity of the charClass rule.

Reply:
This will help to capture a sequence of exactly 1 `charClassIntersection` in the `charClass` rule, the rest of that rule can stay the exact same 😉
```scala
charClassIntersection.rep(exactly = 1)
```

Reply:
Ah nice cheat 👌